src/runtime/PointerValue.h

```diff
@@ -163,6 +156,11 @@ class PointerValue : public gc {
         return hasTag(g_doubleInEncodedValueTag);
     }
 
+    inline bool hasArrayObjectTag() const
+    {
+        return hasTag(g_arrayObjectTag);
```

Comment:
Can we replace `hasArrayObjectTag` with existing `isArrayObject` method?
ArrayPrototypeObject is also ArrayObject.
```c++
    inline bool isArrayObject() const
    {
        return hasTag(g_arrayObjectTag) || hasTag(g_arrayPrototypeObjectTag);
    }
```

Reply:
This is fast path for interpreter. I just want to test with 'g_arrayObjectTag'